napari/utils/perf/_event.py

```diff
@@ -47,3 +64,9 @@ def duration_us(self):
     @property
     def duration_ms(self):
         return self.duration_ns / 1e6
+
+
+class InstantEvent(PerfEvent):
+    def __init__(self, name: str, time_ns: int, **kwargs):
+        super().__init__(name, time_ns, time_ns, **kwargs)
+        self.type = "I"  # instant event
```

Comment:
What's the difference between these two event types, and if there is a class for each, why not use `type(event)`?

Reply:
I added some comments about the different event types, but if you Google for "Trace Event Format" there's a google doc with the full spec, it's about 19 pages.

I'm not sure what you are proposing exactly but in general with OO it's considered much better to ask objects for information like such as `write event.type`, where the object "knows" the right answer, than to externally interrogate their type like `write "I" if type(event) == InstantEvent" else "X"` which is considered more fragile.

That said confining the details of their format to one place in the code would be nice, so maybe it is worth using `type(event)` or an enum to have a translation spot from our code to their format. I think with just "X" and "I" it's not bad as-is, but maybe if we add more even types we should consider that.